subsys/bluetooth/host/classic/hfp_hf.c

```diff
@@ -391,732 +393,2065 @@ int cind_resp(struct at_client *hf_at, struct net_buf *buf)
-				bt_hf->remote_ringing(hf);
-			}
-		}
-		break;
-	case HF_CALL_HELD_IND:
-		if (bt_hf->call_held) {
-			bt_hf->call_held(hf, value);
-		}
-		break;
-	case HF_SINGNAL_IND:
-		if (bt_hf->signal) {
-			bt_hf->signal(hf, value);
-		}
-		break;
-	case HF_ROAM_IND:
-		if (bt_hf->roam) {
-			bt_hf->roam(hf, value);
+		if (call->index == index) {
+			return call;
 		}
-		break;
-	case HF_BATTERY_IND:
-		if (bt_hf->battery) {
-			bt_hf->battery(hf, value);
-		}
-		break;
-	default:
-		LOG_ERR("Unknown AG indicator");
-		break;
 	}
+
+	return NULL;
 }
 
-int cind_status_handle(struct at_client *hf_at)
+static struct bt_hfp_hf_call *get_call_without_index(struct bt_hfp_hf *hf)
 {
-	uint32_t index = 0U;
+	struct bt_hfp_hf_call *call;
 
-	while (at_has_next_list(hf_at)) {
-		uint32_t value;
-		int ret;
+	for (size_t index = 0; index < ARRAY_SIZE(hf->calls); index++) {
+		call = &hf->calls[index];
 
-		ret = at_get_number(hf_at, &value);
-		if (ret < 0) {
-			LOG_ERR("could not get the value");
-			return ret;
+		if (!atomic_test_bit(call->flags, BT_HFP_HF_CALL_IN_USING)) {
+			continue;
 		}
 
-		ag_indicator_handle_values(hf_at, index, value);
-
-		index++;
+		if (!call->index) {
+			return call;
+		}
 	}
 
-	return 0;
+	return NULL;
 }
+#endif /* CONFIG_BT_HFP_HF_ECS */
 
-int cind_status_resp(struct at_client *hf_at, struct net_buf *buf)
+static void hf_reject_call(struct bt_hfp_hf_call *call)
 {
-	int err;
-
-	err = at_parse_cmd_input(hf_at, buf, "CIND", cind_status_handle,
-				 AT_CMD_TYPE_NORMAL);
-	if (err < 0) {
-		LOG_ERR("Error parsing CMD input");
-		hf_slc_error(hf_at);
+	if (bt_hf->reject) {
+		bt_hf->reject(call);
 	}
+	free_call(call);
+}
 
-	return 0;
+static void hf_terminate_call(struct bt_hfp_hf_call *call)
+{
+	if (bt_hf->terminate) {
+		bt_hf->terminate(call);
+	}
+	free_call(call);
 }
 
-int ciev_handle(struct at_client *hf_at)
+static void clear_call_without_clcc(struct bt_hfp_hf *hf)
 {
-	uint32_t index, value;
-	int ret;
+	struct bt_hfp_hf_call *call;
 
-	ret = at_get_number(hf_at, &index);
-	if (ret < 0) {
-		LOG_ERR("could not get the Index");
-		return ret;
-	}
-	/* The first element of the list shall have 1 */
-	if (!index) {
-		LOG_ERR("Invalid index value '0'");
-		return 0;
-	}
+	for (size_t index = 0; index < ARRAY_SIZE(hf->calls); index++) {
+		call = &hf->calls[index];
 
-	ret = at_get_number(hf_at, &value);
-	if (ret < 0) {
-		LOG_ERR("could not get the value");
-		return ret;
-	}
+		if (!atomic_test_bit(call->flags, BT_HFP_HF_CALL_IN_USING)) {
+			continue;
+		}
 
-	ag_indicator_handle_values(hf_at, (index - 1), value);
+		if (atomic_test_and_clear_bit(call->flags, BT_HFP_HF_CALL_CLCC)) {
+			continue;
+		}
 
-	return 0;
+		switch (atomic_get(call->state)) {
+		case BT_HFP_HF_CALL_STATE_TERMINATE:
+			break;
+		case BT_HFP_HF_CALL_STATE_OUTGOING:
+			__fallthrough;
+		case BT_HFP_HF_CALL_STATE_INCOMING:
+			__fallthrough;
+		case BT_HFP_HF_CALL_STATE_ALERTING:
+			__fallthrough;
+		case BT_HFP_HF_CALL_STATE_WAITING:
```

Comment:
You don't need to specify `__fallthrough` if the case statements are on consecutive lines.

Reply:
Updated.